src/test/java/org/opensearch/flowframework/workflow/DeployModelStepTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.ml.common.MLTaskState;
+import org.opensearch.ml.common.MLTaskType;
+import org.opensearch.ml.common.transport.deploy.MLDeployModelResponse;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.client.NoOpNodeClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class DeployModelStepTests extends OpenSearchTestCase {
+
+    private WorkflowData inputData = WorkflowData.EMPTY;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private NodeClient nodeClient;
+
+    @Mock
+    MachineLearningNodeClient machineLearningNodeClient;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        inputData = new WorkflowData(Map.ofEntries(Map.entry("model_id", "modelId")));
+
+        MockitoAnnotations.openMocks(this);
+
+        nodeClient = new NoOpNodeClient("xyz");
+
+    }
+
+    public void testDeployModel() {
+
+        String taskId = "taskId";
+        String status = MLTaskState.CREATED.name();
+        MLTaskType mlTaskType = MLTaskType.DEPLOY_MODEL;
+
+        DeployModelStep deployModel = new DeployModelStep(nodeClient);
+
+        ArgumentCaptor<ActionListener<MLDeployModelResponse>> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+
+        doAnswer(invocation -> {
+            ActionListener<MLDeployModelResponse> actionListener = invocation.getArgument(2);
+            MLDeployModelResponse output = new MLDeployModelResponse(taskId, mlTaskType, status);
+            actionListener.onResponse(output);
+            return null;
+        }).when(machineLearningNodeClient).deploy(eq("modelId"), actionListenerCaptor.capture());
+
+        CompletableFuture<WorkflowData> future = deployModel.execute(List.of(inputData));
+
+        // TODO: Find a way to verify the below
+        // verify(machineLearningNodeClient).deploy(eq(MLRegisterModelInput.class), actionListenerCaptor.capture());
```

Comment:
@dbwiddis any idea here? Approaches I tried:
1 `MockedStatic` with mockito-inline 
2. Created ArgumentCaptor only for `MLDeployModelResponse`.
Although, I can see the method getting invoked in the debugger. Keeping it as a TODO here for now.

Reply:
Hard to answer without live debugging.  In general my debugging suggestion is to try verify with `any(Object.class)` for the arguments.  If this fails, the problem is not in this line, but you're encountering a failure in `deploy()`.   

Toss up a new issue to debug this so we can get this merged.

Reply:
This was the 3rd option I tried and forgot to mention. I'll create a new issue and link it here.

Reply:
Looked into this.  The `machineLearningNodeClient` you are verifying is the local mock from this test class.  However, when you call `deployModel.execute()` you are creating a new client with this line:
```java
MachineLearningNodeClient machineLearningNodeClient = MLClient.createMLClient(nodeClient);
```

Reply:
Thanks for looking into this. Let me create a follow up PR for the above.